lib/parsers/webhook_json_parser.rb

```diff
@@ -27,20 +29,12 @@ def detect_vcs
 
       def github_webhook?
         # https://developer.github.com/v3/activity/events/types/#pushevent
-        # X-GitHub-Event header is set, but not accessible here.
-        return false unless @data.key? 'repository'
-        return false unless @data['repository'].key? 'id'
-        return false unless @data['repository'].key? 'html_url'
-        return false unless @data['repository']['html_url'] =~ %r{github\.com}
-        true
+        @env.key?('HTTP_X_GITHUB_EVENT')
```

Comment:
This is the big win.  It's so much nicer now that we have easy access to the headers.

Reply:
Much safer too, since that's the public API. Relying on internal implementation details always seems so hacky. (not to mention that it would fail on self-hosted GitHub appliances.)